### DISCUSSION TITLE
gh cs ssh not creating keys

### DISCUSSION THREAD

**Discussion by gigamonkey** [community/community#52556] [Codespaces] [answered]
### Select Topic Area

Question

### Body

According to the docs here https://cli.github.com/manual/gh_codespace_ssh:

> By default, the 'ssh' command will create a public/private ssh key pair to authenticate with the codespace inside the ~/.ssh directory.

But when I run `gh cs ssh` it connects fine to my codespace but does not seem to create a key pair in my `~/.ssh` directory. Which is a problem because when I run `gh cs ssh --config` it writes a configuration that references a non-existent `~/.ssh/codespaces.auto.pub` key file.

Am I misunderstanding something about what "create a public/private ssh key pair" means? I'm on a Mac for whatever that's worth.

**Comment by joshaber** [comment, marked as answer]
Hi!

It looks like there are at least a couple things we need to fix here 😅 
1. We need to update the docs. `gh cs ssh` currently only creates SSH keys as a last resort if there are no other keys to use.
2. `gh cs ssh --config` should output the config for the SSH identity it actually uses and not assuming it's the generated identity (which may not even exist!).

I've opened internal issues for both of those.

**Reply by jkeech** [reply]
I opened https://github.com/cli/cli/pull/7325 for (1)

**Comment by gigamonkey** [comment]
Gotcha. So I can had fix the config output. Yeah, that works. Thanks for opening the issues!

**Comment by ericboehs** [comment]
I ended up just symlinking my existing pub/private key to `codespaces.auto`:
```shell
printf '\nMatch all\nInclude ~/.ssh/codespaces\n' >> ~/.ssh/config
gh cs ssh --config > ~/.ssh/codespaces
ln -s ~/.ssh/id_rsa ~/.ssh/codespaces.auto
ln -s ~/.ssh/id_rsa.pub ~/.ssh/codespaces.auto.pub
```

Now I can just `ssh cs<Tab>` to ssh to my codespaces.
